### PR TITLE
Add SecUtf8 for secure valid UTF-8 strings

### LINKED ISSUE #7: Make SecStr newtype that enforces UTF-8
What do you think of making `SecStr` a new-type wrapper that enforces UTF-8, since the name is hinting that it contains a valid `String` and in Rust a valid string is always `UTF-8`. It would be something like:

```rust
pub struct SecStr(secstr::SecVec<u8>);

impl From<String> for SecStr {
    fn from(s: String) -> SecStr {
        SecStr(secstr::SecVec::from(s))
    }
}

impl SecStr {
    pub fn unsecure(&self) -> &str {
        std::str::from_utf8(self.0.unsecure()).expect("cast SecStr to utf8")
    }
}
```

I can submit a PR if you agree with the idea.

### PR DESCRIPTION
**Pull request by dvic**
Fixes #7

### CODE REVIEW

**Comment by dvic** [comment]
Force pushses are because the indentation was completely messed up 😆 I'm used to `cargo fmt` (see #9).

File: src/lib.rs [hunk 1]
```diff
@@ -149,15 +149,89 @@ mod memlock {
...
+                Ok(SecUtf8::from(v.to_string()))
+            }
+        }
+        deserializer.deserialize_string(SecUtf8Visitor)
+    }
+}
 
-/// A data type suitable for storing sensitive information such as passwords and private keys in memory, that implements:  
-/// 
-/// - Automatic zeroing in `Drop`
```
Comment (valpackett) [inline src/lib.rs:155]: The trailing whitespace here — in doc comments specifically — was intentional: two spaces at the end signify a line break (`<br>`). Though this is a bulleted list anyway, I'm not sure this is required…

Comment (dvic) [reply src/lib.rs:155]: I'm not sure, this is how it looks like
screenshot 2019-01-14 at 12 54 18

Looks ok, no?

Comment (valpackett) [reply src/lib.rs:155]: yeah that's fine

**Comment by dvic** [comment]
@myfreeweb Is there anything I should add to this PR? Have you had a chance to review it? Thanks 👍

**Comment by valpackett** [comment]
Looks good, thanks

**Pull request merged**